src/RuleAsset.php

```diff
@@ -269,6 +298,54 @@ public function executeActions($output, $params, array $input = [])
                                     }
                                 }
                                 break;
+                            case "_affect_user_by_email_and_regex":
+                                foreach ($this->regex_results as $regex_result) {
+                                     $res = RuleAction::getRegexResultById(
+                                         $action->fields["value"],
+                                         $regex_result
+                                     );
+                                    if ($res != null) {
+                                          $user = User::getUsersIdByEmails($res);
+                                        if ($user) {
+                                            $output['users_id'] = $user[0];
+                                        } else {
+                                            $output['users_id'] = 0;
+                                        }
+                                    }
+                                }
+                                break;
+                            case "_affect_user_by_registration_number_and_regex":
+                                foreach ($this->regex_results as $regex_result) {
+                                     $res = RuleAction::getRegexResultById(
+                                         $action->fields["value"],
+                                         $regex_result
+                                     );
+                                    if ($res != null) {
+                                          $user = User::getIdByField('registration_number', $res);
+                                        if ($user) {
+                                            $output['users_id'] = $user;
+                                        } else {
+                                            $output['users_id'] = 0;
+                                        }
+                                    }
+                                }
+                                break;
+                            case "_affect_user_by_sync_field_and_regex":
+                                foreach ($this->regex_results as $regex_result) {
+                                        $res = RuleAction::getRegexResultById(
+                                            $action->fields["value"],
+                                            $regex_result
+                                        );
+                                    if ($res != null) {
+                                            $user = User::getIdByField('sync_field', $res);
+                                        if ($user) {
+                                            $output['users_id'] = $user;
+                                        } else {
+                                            $output['users_id'] = 0;
+                                        }
```

Comment:
we should not reset the field to 0

If the correct user is not found, leave the field untouched.